### PR TITLE
[dev] Adding `libnghttp2*` `Provides` to `nghttp2`.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

The binaries we provide through the `nghttp2*` packages are also known under the names `libnghttps*`, so we're adding the `Provides` statements to cover such cases.

###### Change Log  

- Added `libnghttp2*` `Provides` to `nghttp2`.

###### Does this affect the toolchain?  

No.

###### Test Methodology

- Local package build.

### CODE REVIEW

**Review by mfrw — APPROVED** [review]

File: SPECS/nghttp2/nghttp2.spec [hunk 1]
```diff
@@ -58,6 +62,9 @@ rm %{buildroot}/%{_libdir}/*.la
 %{_libdir}/pkgconfig/*.pc
 
 %changelog
+* Tue Nov 03 2020 Pawel Winogrodzki <pawelwi@microsoft.com> - 1.41.0-2
```
Comment (thcrain-msft) [inline SPECS/nghttp2/nghttp2.spec:65]: Can we update this date?

Comment (mfrw) [reply SPECS/nghttp2/nghttp2.spec:65]: Good Catch .. did not even notice the date was incorrect :(

Comment (mfrw) [reply SPECS/nghttp2/nghttp2.spec:65]:
```suggestion
* Wed Jun 23 2021 Pawel Winogrodzki <pawelwi@microsoft.com> - 1.41.0-2
```

Comment (PawelWMS) [reply SPECS/nghttp2/nghttp2.spec:65]: Thanks, guys!

**Pull request merged**